### DISCUSSION TITLE
credential.helper=cache only works before starting lazygit

### DISCUSSION THREAD

**Discussion by rayluo** [jesseduffield/lazygit#1496] [Q&A] [answered]
**Describe the bug**
Background: In some of my environments I'm using a normal password (i.e. not ssh key), and I use `git credential.helper cache` to reduce the need to input password everytime. It works by git spawning a background cache-daemon.

Such mechanism still works, if I perform one such login in normal git command line, and then start to do all the pull/push inside lazygit.
But it will not work if I use lazygit without that cache-daemon being pre-existing. It is possibly because git's cache daemon attaches itself to current process's parent, but perhaps lazygit uses an intermedium and temporary "bash" process so that the daemon got killed?

**Comment by jesseduffield** [comment, marked as answer]
That's right: lazygit either runs git directly (no bash involved) or via a short-lived bash process. Not sure how to get around this one.

**Comment by rayluo** [comment]
Fair enough. Until we would have a real solution, I would suggest @jesseduffield you convert this github issue into a github discussion, and mark it as "answered", so that it would at least become a known workaround for more audience.
